command/eval_list.go

```diff
@@ -0,0 +1,207 @@
+		if err != nil {
+			return []string{}
+		}
+		return resp.Matches[contexts.Evals]
+	})
+}
+
+func (c *EvalListCommand) Name() string { return "eval list" }
+
+func (c *EvalListCommand) Run(args []string) int {
+	var monitor, verbose, json bool
+	var perPage int
+	var tmpl, pageToken, filterJobID, filterStatus string
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.BoolVar(&monitor, "monitor", false, "")
+	flags.BoolVar(&verbose, "verbose", false, "")
+	flags.BoolVar(&json, "json", false, "")
+	flags.StringVar(&tmpl, "t", "", "")
+	flags.IntVar(&perPage, "per-page", 0, "")
+	flags.StringVar(&pageToken, "page-token", "", "")
+	flags.StringVar(&filterJobID, "job", "", "")
+	flags.StringVar(&filterStatus, "status", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments
+	args = flags.Args()
+	if l := len(args); l != 0 {
+		c.Ui.Error("This command takes no arguments")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	opts := &api.QueryOptions{
+		PerPage:   int32(perPage),
+		NextToken: pageToken,
+		Params:    map[string]string{},
+	}
+	if filterJobID != "" {
+		opts.Params["job"] = filterJobID
+	}
+	if filterStatus != "" {
+		opts.Params["status"] = filterStatus
+	}
+
+	evals, qm, err := client.Evaluations().List(opts)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error querying evaluations: %v", err))
+		return 1
+	}
+
+	// If args not specified but output format is specified, format
+	// and output the evaluations data list
+	if json || len(tmpl) > 0 {
+		out, err := Format(json, tmpl, evals)
+		if err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+
+		c.Ui.Output(out)
+		return 0
+	}
+
+	if len(evals) == 0 {
+		c.Ui.Output("No evals found")
+		return 0
+	}
+
+	// Truncate the id unless full length is requested
+	length := shortId
+	if verbose {
+		length = fullId
+	}
+
+	out := make([]string, len(evals)+1)
+	out[0] = "ID|Priority|Triggered By|Job ID|Status|Placement Failures"
+	for i, eval := range evals {
+		failures, _ := evalFailureStatus(eval)
+		out[i+1] = fmt.Sprintf("%s|%d|%s|%s|%s|%s",
+			limit(eval.ID, length),
+			eval.Priority,
+			eval.TriggeredBy,
+			eval.JobID,
+			eval.Status,
+			failures,
+		)
+	}
+	c.Ui.Output(formatList(out))
+
+	if qm.NextToken != "" {
+		c.Ui.Output(fmt.Sprintf(`
+Results have been paginated. To get the next page run:
+
+%s -page-token %s`, argsWithoutPageToken(os.Args), qm.NextToken))
+	}
+
+	return 0
+}
+
+// argsWithoutPageToken strips out of the -page-token argument and
+// returns the joined string
+func argsWithoutPageToken(osArgs []string) string {
+	args := []string{}
+	i := 0
+	for {
+		if i >= len(os.Args) {
+			break
+		}
+		arg := os.Args[i]
+		if arg == "-page-token" {
+			i += 2
+			continue
+		}
```

Comment:
Ahh good call, I forgot to consider that the page token may be in the input command as well.

But I think this needs to be adjusted slightly to consider the case where the token is passed as `-page-token=XYZ`. Something like:

```go
if strings.HasPrefix(arg, "-page-token") {
	if strings.Contains(arg, "=") {
		i += 1
	} else {
		i += 2
	}
	continue
}
```

Reply:
Good catch. There's also a bug that it uses `os.Args` and not the passed in `osArgs`, so I'm writing a test now because "oh this is a tiny little thing no need to write a test" is how we get terrible bugs shipped. 😁